llm-translator.html

```diff
@@ -0,0 +1,191 @@
+          url: ROOM_URL,
+          //startAudioOff: true,
+          //startVideoOff: true,
+          subscribeToTracksAutomatically: false,
+          audioSource: false,
+          videoSource: false,
+          dailyConfig: {},
+        });
+
+        [
+          "participant-updated",
+          "participant-left",
+          "error",
+          "camera-error",
+          "started-camera",
+          "active-speaker-change",
+          "recording-started",
+          "recording-stopped",
+          "recording-error",
+          "track-started",
+          "track-stopped",
+        ].forEach((e) => {
+          call.on(e, (ev) => {
+            console.log(`🔥 ${e}:`, ev);
+          });
+        });
+
+        call.on("track-started", startTrack);
+        call.on("track-stopped", stopTrack);
+
+        call.on("participant-joined", (ev) => {
+          console.log("🤠 Participant joined: ", ev);
+          const p = ev.participant;
+          // is this a person or a translator bot?
+          if (isTranslator(p)) {
+            // This is a translator. Update the list of languages
+            // to include this translator's language
+            console.log("this is a translator");
+            const lang = translatorLanguage(p);
+            const newOpt = document.createElement("option");
+            newOpt.value = lang;
+            newOpt.text = cap(lang);
+            languageSelect.appendChild(newOpt);
+          } else {
+            // This is a human. Subscribe to their video, and their
+            // audio if the current language is English
+            console.log("this is a human");
+            console.log(languageSelect.value);
+            const audioSub = languageSelect.value === "english";
+            call.updateParticipant(p["session_id"], {
+              setSubscribedTracks: {
+                video: true,
+                audio: audioSub,
+              },
+            });
+          }
+        });
+
+        call.on("app-message", (e) => {
+          console.log("Got app message: ", e);
+          const lang = languageSelect.value;
+          if (e.fromId === "transcription" && lang === "english") {
+            console.log("got it for english: ", e.data.text);
+            showSubtitle(e.data.session_id, e.data.text);
+          } else if (e.data.translation_language === lang) {
+            console.log("got it for non-english: ", e.data.translation);
+            showSubtitle(e.data.session_id, e.data.translation);
+          }
+          e.fromId;
+        });
+      }
+
+      function startTrack(evt) {
+        console.log("track started: ", evt);
+        if (evt.track.kind === "audio" && !evt.participant.local) {
+          let audiosDiv = document.getElementById("audios");
+          let audioEl = document.createElement("audio");
+          audiosDiv.appendChild(audioEl);
+          audioEl.style.width = "100%";
+          audioEl.srcObject = new MediaStream([evt.track]);
+          audioEl.play();
```

Comment:
Also, do we need to check for the playability of the given track? Is it guaranteed that a track that is started will be playable?